### PR TITLE
fix: array out-of-bounds issue in `_exitMaintenance`

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to fix an issue in `_exitMaintenance` that the array index could be out-of-bounds 

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes:
* add each change in a bullet point here
* ...

### CODE REVIEW

File: contracts/BSCValidatorSet.sol [hunk 1]
```diff
@@ -1294,9 +1298,10 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
             _misdemeanor(validator);
         }
 
-        // step 3: modify global storage
-        --numOfMaintaining;
-        validatorExtraSet[index].isMaintaining = false;
+        // step 4: clear isMaintaining info
```
Comment (unclezoro) [inline contracts/BSCValidatorSet.sol:1301]: if this is the case, we can move `validatorExtraSet[index].isMaintaining = false;` up without condition.

File: contracts/BSCValidatorSet.sol [hunk 2]
```diff
@@ -1202,7 +1202,7 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
             validator = currentValidatorSet[i].consensusAddress;
 
             // exit maintenance
-            isFelony = _exitMaintenance(validator, i, miningValidatorCount);
+            isFelony = _exitMaintenance(validator, i, miningValidatorCount, false);
```
Comment (unclezoro) [inline contracts/BSCValidatorSet.sol:1205]: should't it be true?

File: contracts/interface/ISlashIndicator.sol [hunk 3]
```diff
@@ -2,7 +2,7 @@ pragma solidity 0.6.4;
 
 interface ISlashIndicator {
     function clean() external;
-    function downtimeSlash(address validator, uint256 count) external;
```
Comment (unclezoro) [inline contracts/interface/ISlashIndicator.sol:5]: I will suggest do not include this in the event `shouldRevert`

File: contracts/BSCValidatorSet.sol (hunk 2 above)
Comment (Pythonberg1997) [reply contracts/BSCValidatorSet.sol:1205]: System tx should be false.

File: contracts/interface/ISlashIndicator.sol (hunk 3 above)
Comment (Pythonberg1997) [reply contracts/interface/ISlashIndicator.sol:5]: This is not an event. It's the function interface

**Review by zzzckck — APPROVED** [review]

**Comment by zzzckck** [comment]
pls open an issue to track the failed UT, better to bring it up.

**Review by NathanBSC — DISMISSED** [review]

**Review by cosinlink — APPROVED** [review]

**Review by unclezoro — APPROVED** [review]

**Pull request merged**